types/webpack-env/index.d.ts

```diff
@@ -103,21 +104,21 @@ declare namespace __WebpackModuleApi {
          * The data will be available at module.hot.data on the new module.
          * @param callback
          */
-        dispose<T>(callback: (data: T) => void): void;
+        dispose<T = any>(callback: (data: T) => void): void;
```

Comment:
I get where you're coming from, and in the end this will come down to a matter of opinion, so I'll just list the reasons for why I think this should be changed.

1. This conforms with the typings of `require` in all the other CommonJS module-loading packages I could find, including [Node](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/924fafffc09cfeb0267573af2c847cdbfcfa464d/types/node/index.d.ts#L89), [SystemJS](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/924fafffc09cfeb0267573af2c847cdbfcfa464d/types/systemjs/index.d.ts#L352) and [RequireJS](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/924fafffc09cfeb0267573af2c847cdbfcfa464d/types/requirejs/index.d.ts#L299), meaning you can reuse code between them and Webpack without type errors.
1. This exact case is listed in [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes) as a "disguised type assertion", which is effectively what this is. The typings can't at compile-time know/assume what `require` returns, and it has (in my opinion) no business forcing you to state it. Just use a type assertion (`require('thing') as IThing`).
1. I would like to be able to `require` a bunch of non-TypeScript entities that don't have typings without having to write `<any>` everywhere. This is especially true for Webpack where you can (and do) `require` assets such as CSS/images/fonts/whatever, which is 99% of my `require` usage in Webpack projects.

This change shouldn't interfere with strict mode as far as I know. I've always had every strict flag turned on and have had no problems using this variant.

I guess @DanielRosenwasser or some other collaborator will have to make a final decision.